redis/src/types.rs

```diff
@@ -1460,17 +1460,50 @@ impl<T: ToRedisArgs> ToRedisArgs for Option<T> {
     }
 }
 
-impl<T: ToRedisArgs> ToRedisArgs for &T {
-    fn write_redis_args<W>(&self, out: &mut W)
-    where
-        W: ?Sized + RedisWrite,
-    {
-        (*self).write_redis_args(out)
-    }
+macro_rules! deref_impl {
+    (
+        $(#[$attr:meta])*
+        <$($desc:tt)+
+    ) => {
+        $(#[$attr])*
+        impl <$($desc)+ {
+            #[inline]
+            fn write_redis_args<W>(&self, out: &mut W)
+                where
+                W: ?Sized + RedisWrite,
+            {
+                (**self).write_redis_args(out)
+            }
 
-    fn is_single_arg(&self) -> bool {
-        (*self).is_single_arg()
-    }
+            fn is_single_arg(&self) -> bool {
+                (**self).is_single_arg()
+            }
+
+            fn describe_numeric_behavior(&self) -> NumericBehavior {
+                (**self).describe_numeric_behavior()
+            }
+        }
+    };
+}
+
+deref_impl! {
+    <'a, T: ?Sized> ToRedisArgs for &'a T where T: ToRedisArgs
+}
+
+deref_impl! {
+    <'a, T: ?Sized> ToRedisArgs for &'a mut T where T: ToRedisArgs
+}
+
+deref_impl! {
+    <T: ?Sized> ToRedisArgs for Box<T> where T: ToRedisArgs
+}
+
+deref_impl! {
+    /// Encoding a data structure containing `Arc` will encode a copy of
```

Comment:
I don't understand this. encoding in this library always incurs a copy of the content, what are you clarifying?

Reply:
you can reuse the output, but this macro is for the inputs :)
`ToRedisArgs` - inputs
`FromRedisValue` - outputs